Write 2018 development proposal

dhimmel opened the pull request:
Here's the PDF, which I built locally, from https://github.com/greenelab/manufund-2018/pull/2/commits/69cb000f4fc2533334ecd2b4fc3e752beaf16b9e: [`manuscript.pdf`](https://github.com/greenelab/manufund-2018/files/1580500/manuscript.pdf)

dhimmel committed:
First draft of proposal

dhimmel committed:
Cite Steemit Bitcoin Whitepaper post

Did not add citation to PeerJ Preprint of the Sci-Hub Coverage
Study [@doi:10.7287/peerj.preprints.3100v2] because DOI Content
Negotiation failed. See https://github.com/datacite/datacite/issues/187#issuecomment-353448091

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+## Proposal {.page_break_before}
+
+Manubot is a system for writing scholarly manuscripts on GitHub.
```
Since you're writing this grant in it, should this be scholarly works?

dhimmel replied:
Is works a bit broad? I thinks datasets, jupyter notebooks, and figures all fall under works. Perhaps "scholarly documents"? Were you envisioning changing just here or everywhere in this proposal?

cgreene replied:
I like `scholarly documents` - I agree works is too broad. I think just here is probably ok. I just wondered if this was a point to note that it was used for this grant. 😄

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/46ce5137a37b6042e3ff5a740b7f0991cc578b23

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+
+Manubot is a system for writing scholarly manuscripts on GitHub.
+Manubot aims to **transform publishing** to be transparent & reproducible, immediate & permissionless, versioned & automated, collaborative & open, linked & provenanced, decentralized & hackable, interactive & annotated, and free of charge.
+Wooh!
```
I'm not sure this helps you out here. 😉

dhimmel replied:
Removed in https://github.com/greenelab/manufund-2018/pull/2/commits/fd5747d4f2979a3fdd3e27450c398c3b4d6d3af4

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+## Proposal {.page_break_before}
+
+Manubot is a system for writing scholarly manuscripts on GitHub.
+Manubot aims to **transform publishing** to be transparent & reproducible, immediate & permissionless, versioned & automated, collaborative & open, linked & provenanced, decentralized & hackable, interactive & annotated, and free of charge.
```
This list is pretty extensive. I think you should make clear that this is a long term goal:
"Our __long term__ goal with manubot is to transform publishing to be..." 
also `free of charge` doesn't specify the type of freedom. Free to read, improve, and mine?

dhimmel replied:
These are not just long-term goals... they are what the Manubot provides today to anyone who decides to use it!

dhimmel replied:
> also "free of charge" doesn't specify the type of freedom. Free to read, improve, and mine?

I just meant that you don't have to pay to use Manubot... anyone can produce publication quality manuscripts without buying any products (Word, EndNote, a journal, etcetera).

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Manubot aims to **transform publishing** to be transparent & reproducible, immediate & permissionless, versioned & automated, collaborative & open, linked & provenanced, decentralized & hackable, interactive & annotated, and free of charge.
+Wooh!
+
+Manubot applies **tactics from open source** software to manuscript writing.
```
Suggest: `Manubot is based on a collection of tools from the open source ecosystem`

dhimmel replied:
This is true, but was not the point I was trying to get across. The point is that we adopt the workflow of OSS:

+ writing is done publicly under an open license
+ additions are subject to review
+ contributions are traceable
+ everything is scripted and automated / continuous integration
+ identifiers

cgreene replied:
Ahh! I missed that. Perhaps this would be clearer:
> Manubot transfers the lessons from the open source software movement to academic writing.

And then you could name some of those features here in addition to talking about the technologies you are using.

dhimmel replied:
Okay how about https://github.com/greenelab/manufund-2018/pull/2/commits/dd761d614e0a9c9f67c909cef9e1ce0a89f44d26?

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Manubot applies **tactics from open source** software to manuscript writing.
+Manuscripts are written in markdown, using git for version control and GitHub for collaboration and review.
+Continuous integration automatically rebuilds and deploys the manuscript when it changes [@doi:10.1038/nbt.3780; @doi:10.1038/550143a].
+Manubot provides many state-of-the-art features.
```
I'd remove this assertion. It's only helpful if you can provide a lot of features as support, which a 1-pager doesn't allow.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/65d24ce1435ad83ef0ab995aa5e7432766465208

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+
+We initially created the Manubot to enable the **Deep Review**, a review paper on deep learning that we collaboratively drafted [on GitHub](https://github.com/greenelab/deep-review).
+Over 30 authors contributed to the Deep Review [@doi:10.1101/142760], using [issues](https://github.com/greenelab/deep-review/issues?utf8=%E2%9C%93&q=is%3Aissue) to discuss source material and [pull requests](https://github.com/greenelab/deep-review/pulls?utf8=%E2%9C%93&q=is%3Apr) to propose manuscript changes.
+The community appreciated the breadth of perspectives on a quickly-developing topic and the Deep Review was the most-viewed _bioRxiv_ preprint from April–June 2017 [@doi:10.1038/d41586-017-08493-x].
```
Check the metrics for the others, but I think it was actually the most-viewed preprint from all of 2017.

dhimmel replied:
> most-viewed preprint from all of 2017

Interesting! Out of the four quarterly winners in the [Nature News article](https://doi.org/10.1038/d41586-017-08493-x "2017 in news: The science events that shaped the year"), the Deep Review has the most abstract and PDF views as of December 21, 2017:

[Regulation of Life Span by the Gut Microbiota in The Short-Lived African Turquoise Killifish](https://www.biorxiv.org/content/early/2017/03/27/120980.article-metrics#details)
12,619 abstract views, 13,502 PDF views

[Opportunities And Obstacles For Deep Learning In Biology And Medicine](https://www.biorxiv.org/content/early/2017/05/28/142760.article-metrics#details)
29,235 abstract views, 22,573 PDF views

[Why Does the Neocortex Have Columns, A Theory of Learning the Structure of the World](https://www.biorxiv.org/content/early/2017/09/28/162263.article-metrics#details)
20,705 abstract views, 17,504 PDF views

[RNA velocity in single cells](https://www.biorxiv.org/content/early/2017/10/19/206052.article-metrics#details)
6,733 abstract views, 3,496 PDF views

I'm not sure whether the selection of the preprints in the Nature News article only counts views during that quarter, or instead counts all views up to query time. If the former, it's possible Deep Review is not the most viewed _bioRxiv_ preprint from 2017.

cgreene replied:
My impression is that it is all views up to query time, although the wording is unclear. I have asked Richard Sever via twitter DM for a clarification.

cgreene replied:
Ok, I have also sent him a link to this exact comment in case he prefers to reply via GitHub! 🤞

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+We initially created the Manubot to enable the **Deep Review**, a review paper on deep learning that we collaboratively drafted [on GitHub](https://github.com/greenelab/deep-review).
+Over 30 authors contributed to the Deep Review [@doi:10.1101/142760], using [issues](https://github.com/greenelab/deep-review/issues?utf8=%E2%9C%93&q=is%3Aissue) to discuss source material and [pull requests](https://github.com/greenelab/deep-review/pulls?utf8=%E2%9C%93&q=is%3Apr) to propose manuscript changes.
+The community appreciated the breadth of perspectives on a quickly-developing topic and the Deep Review was the most-viewed _bioRxiv_ preprint from April–June 2017 [@doi:10.1038/d41586-017-08493-x].
+We've also used the Manubot to write our [Sci-Hub Coverage Study](https://greenelab.github.io/scihub-manuscript/) and to reproduce the [Bitcoin Whitepaper](https://dhimmel.github.io/bitcoin-whitepaper/) [@tag:steemit-post].
```
```
The Sci-Hub coverage study preprint is the second most accessed PeerJ preprint ever (link jordan's resource).
In its first year the collaborative, open, and review-driven manubot authoring process was employed for two of the most accessed preprints of the year.
```
☝️ not fully tied to this, but it's worth noting that I think manubot also helps contributions be better (not just easier to write/update/mine computationally).

dhimmel replied:
Addressed in https://github.com/greenelab/manufund-2018/pull/2/commits/26d1a2eedd4291fbb08cfeab337059f012ad9bce

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+The community appreciated the breadth of perspectives on a quickly-developing topic and the Deep Review was the most-viewed _bioRxiv_ preprint from April–June 2017 [@doi:10.1038/d41586-017-08493-x].
+We've also used the Manubot to write our [Sci-Hub Coverage Study](https://greenelab.github.io/scihub-manuscript/) and to reproduce the [Bitcoin Whitepaper](https://dhimmel.github.io/bitcoin-whitepaper/) [@tag:steemit-post].
+
+We are seeking support to fund Manubot development in 2018.
```
In 2018 we seek to expand manubot's capabilities by accomplishing four feature-driven specific aims and one outreach aim.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/03d7dde28785ff10d0f9acb49f90cd3d411eabd3

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Some **specific aims** that we're excited about are:
+
+1. JATS is the predominant archival format for scholarly manuscripts that is optimized for machine-readability and interoperability.
+We want the Manubot to **export to JATS** and to incorporate an existing **JATS viewer**, such as [Lens](https://lens.elifesciences.org/about/), into Manubot webpages.
```
> We will implement Manubot to **export to JATS** and to incorporate an existing **JATS viewer**, such as [Lens](https://lens.elifesciences.org/about/), into Manubot webpages.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/ffd78fb31da0c87379651eb054df82425a3026ec

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+We want the Manubot to **export to JATS** and to incorporate an existing **JATS viewer**, such as [Lens](https://lens.elifesciences.org/about/), into Manubot webpages.
+
+2. Currently all changes to a manuscript's source are tracked using git.
+However, Manubot does not automatically generate **diffs for rendered outputs**.
```
> We will implement **rendered diff functionality** to show how a manuscript changed from the reader's perspective and make creating "tracked changes" documents for journal resubmission easy.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/ffd78fb31da0c87379651eb054df82425a3026ec

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+By creating rendered diffs we can show how a manuscript changed from the reader's perspective and make creating "tracked changes" documents for journal resubmission easy.
+
+3. **End-to-end reproducibility** is when the provenance of every manuscript result can be traced to its source.
+Manubot allows users to pass template variables, so results can be embedded directly from the analysis that created them.
```
> We will continue to evaluate and improve the Manubot template variables functionality, which allows users to embed results and method parameters directly from the analysis that created them.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/ffd78fb31da0c87379651eb054df82425a3026ec

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Manubot allows users to pass template variables, so results can be embedded directly from the analysis that created them.
+We'd like to experiment more with the templating system and make it as user-friendly as possible.
+
+4. We have several ideas for making **Manubot's citation metadata** system even more powerful and versatile.
```
> The **Manubot citation metadata** system is a powerful and versatile approach for bibliographic information. We will continue to expand Manubot's citation-by-persistent-ID capabilities.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/ffd78fb31da0c87379651eb054df82425a3026ec

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+We're confident that citing (only) by standard identifiers is the way of the future and Manubot can serve as a reference for other applications deciding to implement citation-by-ID.
+
+5. Currently Manubot requires considerable technical expertise [to setup](https://github.com/greenelab/manubot-rootstock/blob/f165f609f33b11fdf71a0db6435d4dd159f23973/SETUP.md) and some technical expertise [to use](https://github.com/greenelab/manubot-rootstock/blob/f165f609f33b11fdf71a0db6435d4dd159f23973/USAGE.md).
+Therefore, the target userbase is researchers or publishers who are already using tools like git and continuous integration for other tasks.
```
> The current userbase consists of researchers or publishers who are using tools like git and continuous integration for other tasks, so we will **improve documentation, develop user stories, and perform outreach activities at relevant conferences**.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/ffd78fb31da0c87379651eb054df82425a3026ec

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Manuscripts are written in markdown, using git for version control and GitHub for collaboration and review.
+Continuous integration automatically rebuilds and deploys the manuscript when it changes [@doi:10.1038/nbt.3780; @doi:10.1038/550143a].
+Manubot provides many state-of-the-art features.
+For example, users cite standard identifiers, while bibliographic details are automatically retrieved.
```
With the removal above: Users can cite standard persistent identifiers and manubot retrieves bibliographic details to automatically create a reference list.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/65d24ce1435ad83ef0ab995aa5e7432766465208

cgreene commented on content/01.proposal.md:
```diff
@@ -0,0 +1,54 @@
+Therefore, the target userbase is researchers or publishers who are already using tools like git and continuous integration for other tasks.
+Another continuing aim is therefore to support **outreach and user adoption**, as well as identifying which potential users will benefit most from the platform.
+
+Our **long-term vision** is an improved system of scholarly communication, brought about by tools such as Manubot.
```
Suggest ordering: 
Our **long-term vision** is an improved ecosystem of scholarly communication.
Imagine you could click an author to highlight only the text they wrote.
Or highlight all results and figures that descend from source code contributions by the selected author.
Or highlight all results that depend on a selected upstream experiment or database.
Or if a new version of software used in the manuscript or an updated dataset is released, rebuild the manuscript with the updated version and see the resulting changes...
regardless of whether you're the original authors or a curious third-party!
Manubot forms the foundation for the traceability of all aspects of a scholarly manuscript, which is the key principle that underlies this potential future.

dhimmel replied:
Done in https://github.com/greenelab/manufund-2018/pull/2/commits/74605fed96e844030034db19f43c0cb3e69c52ca

cgreene reviewed (changes requested):
Some changes requested! Looking exciting!

dhimmel committed:
Mention preprint viewership of Sci-Hub manuscript

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158385379

dhimmel committed:
Rephrase aims introduction

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158385477

dhimmel committed:
Update aims based on Casey's review

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158385572
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158385679
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158386043
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158386422
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158386718

dhimmel committed:
Rephrase long-term vision section

Based on Casey's review
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158387201

dhimmel committed:
Boo, no wooh

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158378834

dhimmel committed:
manuscripts → documents in intro sentence

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158389120

dhimmel committed:
Remove unsupported state-of-the-art feature claim

As suggested by Casey in
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158384365
https://github.com/greenelab/manufund-2018/pull/2#discussion_r158386777

dhimmel committed:
Paragraphs for each OSS lessons and OSS tools

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158398681

cgreene commented on content/01.proposal.md:
```diff
@@ -3,10 +3,15 @@
-Manubot applies **tactics from open source** software to manuscript writing.
+Manubot transfers the lessons from the **open source software movement** to academic writing.
+Documents are written in the open, creating an easy path for readers to become contributors.
+We recommend following a workflow where anyone can propose changes, which are then reviewed by project members and revised as necessary before being incorporated.
```
Cite meta-review here?

dhimmel replied:
We link to the Meta Review below. I was actually thinking we'd want to add something like:

> (for example, [see how](https://github.com/greenelab/manufund-2018/pull/2) this proposal was created)

Which links to this PR. What do you think?

cgreene replied:
Clever! That sounds like a good idea.

cgreene reviewed (approved):
Ok - LGTM 👍 . One suggested citation + the question from richard, but we can update the document if he replies.

dhimmel committed:
Link to this pull request in proposal

https://github.com/greenelab/manufund-2018/pull/2#discussion_r158406483
